### PR TITLE
Improve the first figure tutorial

### PR DESCRIPTION
**Pull request by core-man**
**Description of proposed changes**

As suggested by @weiji14 in https://github.com/GenericMappingTools/pygmt/pull/996#issuecomment-792229893, I will submit my revisions for the documentation of each file in one single PR. After all the revisions posted in #996 are submitted in each PR, I will close it.

This is a revision of [first figure tutorial](https://www.pygmt.org/dev/tutorials/first-figure.html) originally posted in #996.

Related to #983.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/tutorials/first-figure.py [hunk 1]
```diff
@@ -34,11 +35,12 @@
 fig = pygmt.Figure()
 
 ########################################################################################
-# Add elements to the figure using its methods. For example, let's start a map with an
-# automatic frame and ticks around a given longitude and latitude bound, set the
-# projection to Mercator (``M``), and the map width to 8 inches:
+# Add elements to the figure using its methods. For example, let's use
+# :meth:`pygmt.Figure.basemap` to start a map for a region indicated by a given
+# longitude and latitude bound, set the projection to Mercator (**M**), the
+# map width to 8 cm, and frame type to be generated automatically:
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:41]:
```suggestion
# :meth:`pygmt.Figure.basemap` to start a map. We'll use the ``region`` parameter
# provide the longitude and latitude bounds, the ``projection`` parameter to set
# the projection to Mercator (**M**) and the map width to 8cm, and the ``frame``
# parameter to generate a frame with automatic tick and annotation spacing.
```
I find this sentence structure to be easier to read.

File: examples/tutorials/first-figure.py [hunk 2]
```diff
@@ -74,20 +76,25 @@
 # ------------------------------
 #
 # You’ll probably have noticed several things that are different from classic
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:78]:
```suggestion
# You have probably noticed several things that are different from classic
```

File: examples/tutorials/first-figure.py [hunk 3]
```diff
@@ -74,20 +76,25 @@
...
-# command-line GMT. Many of these changes reflect the new GMT modern execution mode that
-# are part of GMT 6. A few are PyGMT exclusive (like the ``savefig`` method).
+# command-line GMT. Many of these changes reflect the new GMT modern execution
+# mode that is part of GMT 6.
 #
 # 1. The name of method is ``coast`` instead of ``pscoast``. As a general rule, all
 #    ``ps*`` modules had their ``ps`` prefix removed. The exceptions are:
 #    ``psxy`` which is now ``plot``, ``psxyz`` which is now ``plot3d``, and ``psscale``
 #    which is now ``colorbar``.
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:85]:
```suggestion
# 1. As a general rule, the ``ps`` prefix has been removed from all ``ps*``
#    modules (PyGMT methods). For example, the name of the GMT 5 module ``pscoast``
#    is ``coast`` in GMT 6 and PyGMT. The exceptions are:  ``psxy`` which is now
#    ``plot``, ``psxyz`` which is now ``plot3d``, and ``psscale`` which is now
#    ``colorbar``.
```

File: examples/tutorials/first-figure.py [hunk 4]
```diff
@@ -74,20 +76,25 @@
...
-# 2. The parameters don't use the GMT 1-letter syntax (**R**, **J**, **B**, etc). We use longer
+#
+# 2. More details can be found at :gmt-docs:`/cookbook/introduction.html#modern-and-classic-mode`
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:87]:
```suggestion
# 2. More details can be found in the :gmt-docs:`GMT cookbook introduction to
#    modern mode </cookbook/introduction.html#modern-and-classic-mode>`.
```
I think descriptions of links provided are helpful.

File: examples/tutorials/first-figure.py [hunk 5]
```diff
@@ -74,20 +76,25 @@
...
+#
+# A few are PyGMT exclusive (like the ``savefig`` method).
+#
+# 1. The parameters don't use the GMT 1-letter syntax (**R**, **J**, **B**, etc). We use longer
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:91]:
```suggestion
# 1. The PyGMT parameters (called options or arguments in GMT) don't use the GMT
#    1-letter syntax (**R**, **J**, **B**, etc). We use longer
```
Might be helpful to specify the difference between GMT and PyGMT terminology.

File: examples/tutorials/first-figure.py [hunk 6]
```diff
@@ -74,20 +76,25 @@
...
 #    aliases for these parameters and have some Python exclusive names. The mapping
 #    between the GMT parameters and their Python counterparts should be straight
 #    forward.
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:94]:
```suggestion
#    between the GMT parameters and their PyGMT aliases should be straightforward.
#    For some modules, these aliases are still being developed.
```

File: examples/tutorials/first-figure.py [hunk 7]
```diff
@@ -74,20 +76,25 @@
...
-# 3. Parameters like ``region`` can take lists as well as strings like ``1/2/3/4``.
-# 4. If a GMT parameter has no options (like ``-B`` instead of ``-Baf``), use a ``True``
+# 2. Parameters like ``region`` can take lists as well as strings like ``1/2/3/4``.
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:95]:
```suggestion
# 2. Parameters like ``region`` can take :class:`lists <list>` as well as strings like ``1/2/3/4``.
```

File: examples/tutorials/first-figure.py [hunk 8]
```diff
@@ -74,20 +76,25 @@
...
+# 3. If a GMT parameter has no options (like ``-B`` instead of ``-Baf``), use a ``True``
 #    in Python. An empty string would also be acceptable. For repeated parameters, such
 #    as ``-B+Loleron -Bxaf -By+lm``, provide a list: ``frame=["+Loleron", "xaf", "y+lm"]``.
```
Comment (meghanrjones) [inline examples/tutorials/first-figure.py:98]:
```suggestion
#    as ``-B+Loleron -Bxaf -By+lm``, provide a :class:`list`: ``frame=["+Loleron", "xaf", "y+lm"]``.
```

**Review by michaelgrund — APPROVED** [review]
Looks good to me!

File: examples/tutorials/first-figure.py [hunk 9]
```diff
@@ -34,11 +35,13 @@
 fig = pygmt.Figure()
 
 ########################################################################################
-# Add elements to the figure using its methods. For example, let's start a map with an
-# automatic frame and ticks around a given longitude and latitude bound, set the
-# projection to Mercator (``M``), and the map width to 8 inches:
+# Add elements to the figure using its methods. For example, let's use
+# :meth:`pygmt.Figure.basemap` to start a map. We'll use the ``region`` parameter
+# provide the longitude and latitude bounds, the ``projection`` parameter to set
```
Comment (seisman) [inline examples/tutorials/first-figure.py:40]:
```suggestion
# to provide the longitude and latitude bounds, the ``projection`` parameter to set
```

File: examples/tutorials/first-figure.py [hunk 10]
```diff
@@ -34,11 +35,13 @@
...
+# the projection to Mercator (**M**) and the map width to 8 cm, and the ``frame``
```
Comment (seisman) [inline examples/tutorials/first-figure.py:41]:
```suggestion
# the projection to Mercator (**M**) and the map width to 15 cm, and the ``frame``
```

File: examples/tutorials/first-figure.py [hunk 11]
```diff
@@ -34,11 +35,13 @@
...
+# parameter to generate a frame with automatic tick and annotation spacings.
 
-fig.basemap(region=[-90, -70, 0, 20], projection="M8i", frame=True)
+fig.basemap(region=[-90, -70, 0, 20], projection="M8c", frame=True)
```
Comment (seisman) [inline examples/tutorials/first-figure.py:44]: The default settings (e.g., annotation font size, frame width) look nicer for 15-cm-wide maps.
```suggestion
fig.basemap(region=[-90, -70, 0, 20], projection="M15c", frame=True)
```

File: examples/tutorials/first-figure.py [hunk 12]
```diff
@@ -56,7 +59,7 @@
 # without calling :meth:`gmt.Figure.basemap`:
 
 fig = pygmt.Figure()
-fig.coast(shorelines=True, region=[-90, -70, 0, 20], projection="M8i", frame=True)
+fig.coast(shorelines=True, region=[-90, -70, 0, 20], projection="M8c", frame=True)
```
Comment (seisman) [inline examples/tutorials/first-figure.py:62]:
```suggestion
fig.coast(shorelines=True, region=[-90, -70, 0, 20], projection="M15c", frame=True)
```

File: examples/tutorials/first-figure.py [hunk 13]
```diff
@@ -73,21 +76,29 @@
...
+#    1-letter syntax (**R**, **J**, **B**, etc). We use longer
 #    aliases for these parameters and have some Python exclusive names. The mapping
-#    between the GMT parameters and their Python counterparts should be straight
-#    forward.
-# 3. Parameters like ``region`` can take lists as well as strings like ``1/2/3/4``.
-# 4. If a GMT parameter has no options (like ``-B`` instead of ``-Baf``), use a ``True``
+#    between the GMT parameters and their PyGMT aliases should be straightforward.
+#    For some modules, these aliases are still being developed.
+# 2. Parameters like ``region`` can take :class:`lists <list>` as well as strings like ``1/2/3/4``.
+# 3. If a GMT parameter has no options (like ``-B`` instead of ``-Baf``), use a ``True``
```
Comment (seisman) [inline examples/tutorials/first-figure.py:100]:
```suggestion
# 3. If a GMT option has no arguments (like ``-B`` instead of ``-Baf``), use a ``True``
```

**Pull request merged**